sources/api/netdog/src/net_config/devices/bonding.rs

```diff
@@ -0,0 +1,184 @@
+use super::{error, Dhcp4ConfigV1, Dhcp6ConfigV1, Result, RouteV1, StaticConfigV1, Validate};
+use super::{validate_addressing, HasDhcp, HasRoutes, HasStatic};
+use crate::interface_name::InterfaceName;
+use serde::Deserialize;
+use snafu::ensure;
+use std::convert::TryFrom;
+use std::net::IpAddr;
+
+#[derive(Debug, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub(crate) struct NetBondV1 {
```

Comment:
What if we adopted the convention to have the "current" version of the struct un-suffixed and only add the suffix when V2 exists? Just a thought to make things a little cleaner. @zmrow ?

```suggestion
pub(crate) struct NetBond {
```

Reply:
It's pretty consistent at this point to have `V1` on the end of these types in netdog code. It would be a bit of a refactor to make that consistent going forward. If we do want to make this change, I highly recommend we do that in a follow on PR because that would increase the surface of this change well outside the original scope and this is already a pretty big PR.

Reply:
I do like the explicit nature of a versioned struct, especially in this context where there are quite a few structs to keep track of.

If there are strong feelings about this, I'd prefer to do that refactor in a separate PR as @yeazelm mentioned.  There's already _alot_ of change here.

Reply:
I think I'll punt this for a topic we can sort out in a later PR but not this one.